src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1525,68 +1534,461 @@ static void LoadRegPair(BYTE* p, int reg1, int reg2, UINT32 offset)
+        }
+
+        codeSize += indirectionsDataSize ? 4 : 0; // auipc
+
+        if (pLookup->testForNull)
+        {
+            codeSize += 12; // ori-beq-jalr
+
+            //padding for 8-byte align (required by EmitHelperWithArg)
+            if (codeSize & 0x7)
+                codeSize += 4;
+
+            codeSize += 32; // size of EmitHelperWithArg
+        }
+        else
+        {
+            codeSize += 4; /* jalr */
+        }
+
+        // the offset value of data_label.
+        uint dataOffset = codeSize;
+
+        codeSize += indirectionsDataSize;
+
+        BEGIN_DYNAMIC_HELPER_EMIT(codeSize);
+
+        BYTE * old_p = p;
+
+        if (indirectionsDataSize)
+        {
+            // auipc  t4, 0
+            *(DWORD*)p = 0x00000e97;
+            p += 4;
+        }
+
+        if (pLookup->testForNull || pLookup->sizeOffset != CORINFO_NO_SIZE_CHECK)
+        {
+            // mv  t2, a0
+            *(DWORD*)p = 0x00050393;
+            p += 4;
+        }
+
+        BYTE* pBLECall = NULL;
+
+        for (WORD i = 0; i < pLookup->indirections; i++)
+        {
+            if (i == pLookup->indirections - 1 && pLookup->sizeOffset != CORINFO_NO_SIZE_CHECK)
+            {
+                _ASSERTE(pLookup->testForNull && i > 0);
+
+                if (pLookup->sizeOffset > 2047)
+                {
+                    // auipc  t4, 0
+                    *(DWORD*)p = 0x00000297; p += 4;
+                    // lw  t4, dataOffset(t4)
+                    *(DWORD*)p = 0x000eae83 | (dataOffset << 20); p += 4;
+                    // add  t5, a0, t4
+                    *(DWORD*)p = 0x01d50f33; p += 4;
+                    // ld  t5, 0(t5)
+                    *(DWORD*)p = 0x000f3f03; 
+
+                    // move to next indirection offset data
+                    dataOffset = dataOffset - 16 + 4; // subtract 16 as we have moved PC by 16 and add 4 as next data is at 4 bytes from previous data
+                }
+                else
+                {
+                    // ld  t5, #(pLookup->sizeOffset)(a0)
+                    *(DWORD*)p = 0x00053f03 | ((UINT32)pLookup->sizeOffset << 20); p += 4;
+                    dataOffset -= 4; // subtract 4 as we have moved PC by 4
+                }
+                // lui  t4, (slotOffset&0xfffff000)>>12
+                *(DWORD*)p = 0x00000eb7 | ((((UINT32)slotOffset & 0xfffff000) >> 12) << 12); p += 4;
+                // addi  t4, t4, (slotOffset&0xfff)
+                *(DWORD*)p = 0x000e8e93 | ((slotOffset & 0xfff) << 20); p += 4;
+                dataOffset -= 8;
+                // blt  t4, t5, CALL HELPER
+                pBLECall = p;       // Offset filled later
+                *(DWORD*)p = 0x01eec063; p += 4;
+                dataOffset -= 4;
+            }
+
+            if(pLookup->offsets[i] > 2047)
+            {
+                _ASSERTE(dataOffset < 2047);
+                // auipc  t4, 0
+                *(DWORD*)p = 0x00000297;
```

Comment:
@shushanhf it seems that for loongarch you also either need to add `pcaddi` for `if(pLookup->offsets[i] > 2047)` case (becase `dataOffset` is relative to current instruction), or change `dataOffset` to relative to the beginning of whole code block.

Reply:
The LA64 had done this.

Reply:
https://github.com/dotnet/runtime/blob/18cac43e79503cb2d9424dd19822e7ee8444db26/src/coreclr/vm/loongarch64/stubs.cpp#L1883-L1891

Liking these, LA64 is ok

Reply:
@gbalykov  What should I do ? I push a new PR to fix this  or  update it within this PR ?

Reply:
I think you can do it in separate PR specific to loongarch

Reply:
Thanks very much !